api-server/api-server-common/src/storage/impls/in_memory/mod.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod transactional;
+
+pub const CURRENT_STORAGE_VERSION: u32 = 1;
+
+use std::collections::BTreeMap;
+
+use common::{
+    chain::{Block, ChainConfig, GenBlock, SignedTransaction, Transaction},
+    primitives::{BlockHeight, Id},
+};
+
+use crate::storage::storage_api::{block_aux_data::BlockAuxData, ApiStorageError};
+
+pub struct ApiInMemoryStorage {
+    block_table: BTreeMap<Id<Block>, Block>,
+    block_aux_data_table: BTreeMap<Id<Block>, BlockAuxData>,
+    main_chain_blocks_table: BTreeMap<BlockHeight, Id<Block>>,
+    transaction_table: BTreeMap<Id<Transaction>, (Id<Block>, SignedTransaction)>,
+    best_block: (BlockHeight, Id<GenBlock>),
+    storage_version: Option<u32>,
+}
+
+impl ApiInMemoryStorage {
+    pub fn new(chain_config: &ChainConfig) -> Self {
+        Self {
+            block_table: BTreeMap::new(),
+            block_aux_data_table: BTreeMap::new(),
+            main_chain_blocks_table: BTreeMap::new(),
+            transaction_table: BTreeMap::new(),
+            best_block: (0.into(), chain_config.genesis_block_id()),
+            storage_version: None,
+        }
+    }
+
+    fn is_initialized(&self) -> Result<bool, ApiStorageError> {
+        let storage_version_handle = self.storage_version;
+        Ok(storage_version_handle.is_some())
+    }
+
+    fn get_block(&self, block_id: Id<Block>) -> Result<Option<Block>, ApiStorageError> {
+        let block_result = self.block_table.get(&block_id);
+        let block = match block_result {
+            Some(blk) => blk,
+            None => return Ok(None),
+        };
+        Ok(Some(block.clone()))
```

Comment:
Minor coding suggestion. This could be reduced to:
```rust
Ok(self.block_table.get(&block_id).cloned())
```

There are multiple occurrences of this pattern in this file.

Reply:
I do have the preference to have a clean value of the result at the end. Helps with debugging.